tensorboard/summary/writer/event_file_writer.py

```diff
@@ -190,6 +203,14 @@ def close(self):
                     self._writer.flush()
                     self._writer.close()
 
+    def _check_worker_status(self):
+        """Makes sure the worker thread is still running and raises exception
+        thrown in the worker thread otherwise.
+        """
+        exception = self._worker.exception
```

Comment:
This seems slightly susceptible to race conditions (since the `exception` variable is set within the background thread). Would checking `self._worker.is_alive()` work here?

Reply:
What race condition do you have in mind? Assignments in Python are atomic ([PEP 583](https://peps.python.org/pep-0583/#atomic-unordered-assignments)), so if the statement `self.exception = ex` has been executed, this line should get the assigned value. I created a local variable mostly for the type checking (if it's performed) to be happy, since the field might be changed theoretically but the local variable cannot. If you're referring to the possibility that the exception has happened but the assignment hasn't been reached, my understanding is that we cannot distinguish it from the background thread being inside `write` function at the moment of the check

If the `write` method proceeds further after this check while this variable is set in a background thread, then it will either put an item into the queue if there's space and then fail or block until the background threads pops an item and then fail nevertheless

If I'm missing a scenario you had in mind, could you please share more details?

Reply:
No, that explanation is sufficient. Thanks!